### PR TITLE
fix gap detection

### PR DESCRIPTION
**Pull request by prolic**
resolves https://github.com/prooph/pdo-event-store/issues/243

/cc @hvanoch

### CODE REVIEW

File: src/Projection/PdoEventStoreProjector.php [hunk 1]
```diff
@@ -551,16 +551,14 @@ public function run(bool $keepRunning = true): void
                 }
 
                 if ($gapDetected && $this->gapDetection) {
-                    $sleep = $this->gapDetection->getSleepForNextRetry();
-
-                    \usleep($sleep);
+                    \usleep($this->gapDetection->getSleepForNextRetry() * 1000);
```
Comment (codeliner) [inline src/Projection/PdoEventStoreProjector.php:554]: I'm thinking about whether we should do the calculation or change the documentation and tell people that sleep time is configured in microseconds :thinking: 

When multiplying sleep time 1 ms is the shortest sleep interval possible. Maybe that's too long for some use cases?

**Review by codeliner — CHANGES_REQUESTED** [review]

File: src/Projection/PdoEventStoreProjector.php (hunk 1 above)
Comment (codeliner) [reply src/Projection/PdoEventStoreProjector.php:554]: The good thing is, changing documentation wouldn't change the current behavior. Gap detection is used in production systems, so people have already identified the best sleep time settings for their projections.

Comment (fritz-gerneth) [reply src/Projection/PdoEventStoreProjector.php:554]: I'm with @codeliner here - afaik this is a BC (as unfortunate it may be).

**Comment by prolic** [comment]
Agreed.

**Pull request closed without merging**